packages/teleport/src/services/upgradeWindow/upgradeWindow.ts

```diff
@@ -0,0 +1,40 @@
+/*
+Copyright 2022 Gravitational, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+import cfg from 'e-teleport/config';
```

Comment:
we can't import things from `e-teleport` in `teleport` b/c `e-teleport` is not a dependency. so we either move the enterprise URL endpoint to OS repo (which i'm not sure if that's okay to do), OR enterprise will need to have it's own `Support` component (which is okay to duplicate)

I would ask for advice on how to proceed on this.  i've seen components handling both OS and enterprise, and if it's trivial enough, it can be in OS, but i also know that we want more things to be in enterprise as well so I'm unsure how to proceed :woman_shrugging: maybe @zmb3 ?

Reply:
Hmm right, thanks.

I think we should keep the URL endpoint in the enterprise.

We can add a `children` props to `Support`, then `webapps.e` has its own `Support` that uses OSS:

SupportE:

```jsx
// packages/webapps.e/teleport/src/Support/Support.tsx
export default function Container(): JSX.Element {
  const ctx = useTeleport();
  // const cluster = ctx.storeUser.state.cluster;
  const upgradeWindowsState = useUpgradeWindowStart(ctx);

  return <SupportE {...upgradeWindowsState} isCloud={cfg.isCloud} />;
}

export const SupportE = ({
  isCloud,
  showScheduleUpgrade,
  scheduleUpgradesVisible,
  closeScheduleUpgrade,
  onUpdate,
  selectedUpgradeWindowStart,
  setSelectedUpgradeWindowStart,
  attempt,
  ...props
}: Props) => {
  return (
    <>
      <Support {...props}>
        {isCloud && (
          <DataContainer title="Scheduled Upgrades">
            <DataItem
              title="Window Start Time"
              data={
                <span>
                  {selectedUpgradeWindowStart}
                  <EditLink onClick={showScheduleUpgrade} ml="2" />
                </span>
              }
            />
            <Text>
              Window Start Time is the hour in which an upgrade may begin.
              Changing this value changes it for everyone in your organization.
            </Text>
          </DataContainer>
        )}
        {scheduleUpgradesVisible && (
          <ScheduleUpgrades
            onSave={onUpdate}
            onCancel={closeScheduleUpgrade}
            selectedWindow={selectedUpgradeWindowStart}
            onSelectedWindowChange={setSelectedUpgradeWindowStart}
            attempt={attempt}
          />
        )}
      </Support>
    </>
  );
};
```

Support OSS:

```jsx
// packages/teleport/src/Support/Support.tsx
export default function Container({ children }): JSX.Element {
  const ctx = useTeleport();
  const cluster = ctx.storeUser.state.cluster;
  // no more upgrade window stuff
  return (
    <Support
      {...cluster}
      isEnterprise={cfg.isEnterprise}
      tunnelPublicAddress={cfg.tunnelPublicAddress}
      isCloud={cfg.isCloud}
      children={children}
    />
  );
}
export const Support = ({
  clusterId,
  authVersion,
  publicURL,
  isEnterprise,
  tunnelPublicAddress,
  isCloud,
  children,
}: Props) => {
// ...
return (
    <FeatureBox pt="4">
      ...
      {children}
    </FeatureBox>
```

This way all the upgrade window stuff is moved to enterprise, and we don't have to duplicate code. The downside is that is kind of strange to have `Support` receiving children, it can be confusing.

Reply:
i think this is fine to me, we did something similar before but i don't remember which component